### PR TITLE
Request for code review

### PR DESCRIPTION
**Pull request by eapenzacharias**

### CODE REVIEW

File: blocks.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        end
+    end
+    def my_none?
+        self.my_each do |value|
+             if yield(value)
+                 return false
+             end
+        end
+    end
+    def my_count
```
Comment (mcrd25) [inline blocks.rb]: Your `my_count` method should also consider when block not given
It should also consider when arg is passed. i.e count would only count if array item == arg
so the method could be like
```ruby
def my_count(arg=nil)
  counter = 0
  if block_given?
    self.my_each do |value|
           if yield(value)
               count+=1
           end
     end
  else
   # do stuff if arg passed
  end
  count
end
```

Comment (mcrd25) [reply blocks.rb]: also if neither arg  or block given it should return the length of array etc.

Comment (mcrd25) [reply blocks.rb]: Please address these comments!

File: blocks.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        end
+    end
+    def my_select
+        result = []
+        self.my_each do |x|
+            result << x if yield(x)
+        end
+        result
+    end
+    def my_all?
```
Comment (mcrd25) [inline blocks.rb]: Hey, this is a great attempt, but it gives the wrong result in code review tests.
Here's why:
it only returns false but never has an occurrence for returning true.

You can do this by simply adding `true` before the method or adding `return true`

File: blocks.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        self.my_each do |value|
+             unless yield(value)
+                 return false
+             end
+        end
+    end
+    def my_any?
+        self.my_each do |value|
+             if yield(value)
+                 return true
```
Comment (mcrd25) [inline blocks.rb]: Similar issue as with `my_all` only ever returns true. What about false? This causes the method to fail tests

File: blocks.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+             end
+        end
+    end
+    def my_none?
```
Comment (mcrd25) [inline blocks.rb]: SAme as with `my_all?`

**Review by mcrd25 — COMMENTED** [review]
# Wow great attempt! A couple of changes till 🚀 
Hi Eapen 👋,

## Concerning project specifications
**This is where you stand:**
- [x] should have working my_each 
- [x] should have working my_each_with_index
- [x] should have working my_select 
- [x] should have working my_all? 
- [x] should have working my_none? 
- [x] should have working my_count 
- [x] should have working my_map
- [x] should have working my_inject 

## Concerning crucial requirements that make project work

**This is where you stand**
- [x] Verify if the project should be done solo or collaboratively and that the commit history reflects that
- [x] The project has a descriptive README.md

## Review Comments

Check out my review comments, please address them to make your project work! Also, please recognize that all comments are to be addressed except for those that are listed as suggested.

**Really great project so far!** 😄 

## Additional Suggestions
- I noticed your indentation is pretty wide, maybe try to stick with 2 spaces 😅 

***If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media.***

**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Wow great attempt! A couple of changes till 🚀 
Hi Eapen 👋,

## Concerning project specifications
**This is where you stand:**
- [x] should have working my_each 
- [x] should have working my_each_with_index
- [x] should have working my_select 
- [x] should have working my_all? 
- [x] should have working my_none? 
- [x] should have working my_count 
- [x] should have working my_map
- [x] should have working my_inject 

## Concerning crucial requirements that make project work

**This is where you stand**
- [x] Verify if the project should be done solo or collaboratively and that the commit history reflects that
- [x] The project has a descriptive README.md

## Review Comments

Check out my review comments, please address them to make your project work! Also, please recognize that all comments are to be addressed except for those that are listed as suggested.

**Really great project so far!** 😄 

## Additional Suggestions
- I noticed your indentation is pretty wide, maybe try to stick with 2 spaces 😅 

***If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media.***

**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/)

File: blocks.rb (hunk 2 above)
Comment (eapenzacharias) [reply blocks.rb]: Pure carelessness. Sorry for that

**Review by AdnanAfsari — APPROVED** [review]
Hello @eapenzacharias @mcrd25 Congratulations 👍 

Nice work , looks good, You can merge the branch now.

Happy Coding :)

**Pull request merged**